crates/counter-agg/src/range.rs

```diff
@@ -0,0 +1,194 @@
+use std::cmp::{min, max};
+use serde::{Deserialize, Serialize};
+
+
+// we always store ranges as half open, inclusive on left, exclusive on right, 
+// we are a discrete type so translating is simple [), this enforces equality
+// between ranges like [0, 10) and [0, 9]
+// None values denote infinite bounds on that side
+#[derive(Debug, PartialEq, Copy, Clone, Serialize, Deserialize)]
+pub struct I64Range {
+    pub left: Option<i64>,
+    pub right: Option<i64>
+}
+
+impl I64Range {
+    pub fn has_infinite(&self)-> bool{
+        self.left.is_none() || self.right.is_none()
+    }
+    
+    pub fn is_valid(&self) -> bool{
+        match (self.left, self.right) {
+            (None, _) => true,
+            (_, None) => true,
+            (Some(a), Some(b)) => a <= b, 
```

Comment:
```suggestion
            (Some(a), Some(b)) => a <= b, 
            _ => true,
```
Suggestion applies to the following functions as well.

Edit: suggestion is to drop the first two cases of the match in favor of the new catchall, doesn't look like github is properly capturing that.

Reply:
Did this for this one and the next, but left it alone for a couple others where I thought it was more clear what the logic was in the other way.